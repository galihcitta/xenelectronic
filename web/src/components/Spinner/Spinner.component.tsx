import React from 'react';

import { Loader } from './Spinner.styles';

export const Spinner: React.FC = () => <Loader data-testid="Spinner" />;
