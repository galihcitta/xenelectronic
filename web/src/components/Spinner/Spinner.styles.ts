import styled from 'styled-components';

import { color } from '../../utils/styles';

export const Loader = styled.span`
  display: block;
  width: 50px;
  height: 50px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 2em;

  &:after {
    content: ' ';
    display: block;
    margin: 0 auto;
    width: 28px;
    height: 28px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid ${color.black};
    border-color: ${color.black} transparent ${color.black} transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }

  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
