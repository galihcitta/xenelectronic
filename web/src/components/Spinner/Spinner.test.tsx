import React from 'react';
import { render } from '@testing-library/react';
import { Spinner } from './Spinner.component';

describe('<Spinner />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
