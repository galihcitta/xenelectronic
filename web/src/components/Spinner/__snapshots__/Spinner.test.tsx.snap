// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bdVaJa dpPFGb"
    data-testid="Spinner"
  />
</DocumentFragment>
`;
